src/memspaces/memspace_host.c

```diff
@@ -0,0 +1,66 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include <assert.h>
+#include <hwloc.h>
+#include <stdlib.h>
+
+#include "../memory_targets/memory_target_numa.h"
+#include "../memspace_internal.h"
+#include "memspace_host_internal.h"
+#include "memspace_numa.h"
+
+umf_result_t umfMemspaceHostCreate(umf_memspace_handle_t *hMemspace) {
+    if (!hMemspace) {
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    umf_result_t umf_ret = UMF_RESULT_SUCCESS;
+
+    hwloc_topology_t topology;
+    if (hwloc_topology_init(&topology)) {
```

Comment:
@pbalcer I know we wanted to strip libumf from malloc symbols (hence using base_alloc for internal allocations) but we have not control over what hwloc calls internally (I think they do call malloc). Will linking with hwloc be a problem for the 'proxy library'?

Reply:
I checked that they call malloc (https://github.com/open-mpi/hwloc/blob/eeeda4ad2d0511a2334a02141dab6138b1e0ce47/hwloc/topology.c). If we want to use HOST memspace in the proxy lib that exports malloc symbol, then we'll go into a recursion here